public/app/features/alerting/unified/hooks/useAlertmanagerConfig.ts

```diff
@@ -1,26 +1,28 @@
+import { SerializedError } from '@reduxjs/toolkit';
 import { useEffect } from 'react';
 
-import { useDispatch } from 'app/types';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 
-import { fetchAlertManagerConfigAction } from '../state/actions';
-import { initialAsyncRequestState } from '../utils/redux';
+type Options = {
+  refetchOnFocus?: boolean;
+  refetchOnReconnect?: boolean;
+};
 
-import { useUnifiedAlertingSelector } from './useUnifiedAlertingSelector';
-
-export function useAlertmanagerConfig(amSourceName?: string) {
-  const dispatch = useDispatch();
+// TODO refactor this so we can just call "alertmanagerApi.endpoints.getAlertmanagerConfiguration" everywhere
```

Comment:
Is the `TODO` still relevant?

Reply:
I'm sort of on the fence about this one, on the one hand it's nice to have abstractions like this hook on the other hand they eventually become something weird and more weird code branches start to get added to it.

For example, this hook only exposes `data` and not `currentData`, sometimes you want one and sometimes the other but this hook doesn't account for it. We can certainly add it but as you can see the abstraction start to become very blurry at that point.